web/web.go

```diff
@@ -526,38 +529,193 @@ func dropForeignKey(w http.ResponseWriter, r *http.Request) {
+		return
+	}
+
+	//check new name for non usage before in another table, foreign key constraint or index
+	if status, err := canRename(newNames, table); !status {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	sp := sessionState.conv.SpSchema[table]
+
+	// Update session with renamed secondary indexes
+	newIndexes := []ddl.CreateIndex{}
+	for _, index := range sp.Indexes {
+		if newName, ok := changeMap[index.Name]; ok {
+			index.Name = newName
+		}
+		newIndexes = append(newIndexes, index)
+	}
+	sp.Indexes = newIndexes
+
+	sessionState.conv.SpSchema[table] = sp
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(sessionState.conv)
+}
+func addIndexes(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+	}
+
+	newIndexes := []ddl.CreateIndex{}
+	if err = json.Unmarshal(reqBody, &newIndexes); err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	//check new name for spanner name validity
+	newNames := getNameSlice(newIndexes)
+	if validity, invalidNewNames := checkSpannerNamesValidity(newNames); !validity {
+		http.Error(w, fmt.Sprintf("New name validation fails for following : %v", invalidNewNames), http.StatusBadRequest)
 		return
 	}
-	if !isUniqueName(newName) {
-		http.Error(w, fmt.Sprintf("New name %s is used before in the schema", newName), http.StatusBadRequest)
+
+	//check new name for non usage before in another table, foreign key constraint or index
+	if status, err := canRename(newNames, table); !status {
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	sp.Fks[position].Name = newName
+
+	sp := sessionState.conv.SpSchema[table]
+	sp.Indexes = append(sp.Indexes, newIndexes...)
+
 	sessionState.conv.SpSchema[table] = sp
 	updateSessionFile()
 	w.WriteHeader(http.StatusOK)
 	json.NewEncoder(w).Encode(sessionState.conv)
 }
 
+func getValuesAsLowerCaseSlice(input map[string]string) []string {
+	output := []string{}
+	for _, value := range input {
+		output = append(output, strings.ToLower(value))
+	}
+	return output
+}
+
+func getNameSlice(input []ddl.CreateIndex) []string {
+	output := make([]string, len(input))
+	for _, value := range input {
+		output = append(output, value.Name)
+	}
+	return output
+}
+
+func checkSpannerNamesValidity(input []string) (bool, []string) {
+	status := true
+	invaliNewNames := []string{}
+	for _, newName := range input {
+		if _, status := internal.FixName(newName); status {
+			status = false
+			invaliNewNames = append(invaliNewNames, newName)
+		}
+	}
+	return status, invaliNewNames
+}
+
+func canRename(names []string, table string) (bool, error) {
+
+	namesMap := map[string]bool{}
+	//Check for this name isn't used by another table
```

Comment:
Consider rewording this as:

// Check that this name isn't already used by another table.

Similarly for other comments below.